t/27-consoles-s3270.t

```diff
@@ -0,0 +1,89 @@
+$ipc_run_mock->redefine(pump => 1);
+
+ok my $s3270_console = consoles::s3270->new('s3270', undef), 's3270_console connection can be established';
+
+subtest 's3270_console start' => sub {
+    my $bless_obj = bless({KIDS => [{VAL => '', PID => '0', NUM => 1, TYPE => 'cmd', RESULT => 1, OPS => []}]}, 'IPC::Run');
+    $ipc_run_mock->redefine(start => sub ($self, $in, $out, $err) {
+            $$out = "success\nconnet($bmwqemu::vars{ZVM_HOST})\nstart to execute process\nok";
+            return $bless_obj;
+    });
+    $s3270_console->start();
+    is_deeply $s3270_console->{connection}, $bless_obj, 's3270 console can be started';
+};
+
+subtest 's3270 send' => sub {
+    my $ret = $s3270_console->send_3270("Connect($bmwqemu::vars{ZVM_HOST})", command_status => 'ok');
+    is $s3270_console->{in}, "Connect($bmwqemu::vars{ZVM_HOST})\n", 'input command matches';
+    is $ret->{terminal_status}, "start to execute process", 'terminal status matches';
+    is $ret->{command_status}, "ok", 'command status matches';
+    is_deeply $ret->{command_output}, ['success', "connet($bmwqemu::vars{ZVM_HOST})"], 'command output matches';
+};
+
+subtest 's3270_console activate' => sub {
+    my $s3270_console_mock = Test::MockModule->new('consoles::s3270');
+    $s3270_console_mock->redefine(send_3270 => sub ($self, $command = '', %arg) {
+            return {'command_output' => ['success'], 'command_status' => 'ok', 'terminal_status' => "Connection to C($bmwqemu::vars{ZVM_HOST}) OK"} unless $command =~ /Wait\([0-9],Output\)/;
+            return {'command_status' => 'ok'} if $command =~ /\b0\b/;
+            return {'command_output' => ['Wait: Timed out'], 'command_status' => 'any'};
+    });
+    combined_like {
+        throws_ok { $s3270_console->activate() } qr/expect_3270:/, 's3270_console connection time out'
+    } qr/expect_3270/, 'result empty';
+};
+
+subtest 's3270_console connect_and_login' => sub {
+    my $count = 0;
+    my $s3270_console_mock = Test::MockModule->new('consoles::s3270');
+    $s3270_console_mock->redefine(send_3270 => sub ($self, $command = '', %arg) {
+            # if ($command =~ /Connect\(\w.+\)/) {
+            return {terminal_status => "Connection to C($bmwqemu::vars{ZVM_HOST}) OK"} if $command =~ /Connect\(\w.+\)/;
+            # }
+    });
+    $s3270_console_mock->redefine(expect_3270 => sub ($self, %arg) {
+            my $return_lines;
+            if ((keys %arg == 1) && exists $arg{buffer_ready}) {
+                $return_lines = ['RECONNECT'] if $count == 1;
+                $return_lines = ['CONNECTED'] if $count == 2;
+            }
+            else {
+                $return_lines = ['Fill in your USERID and PASSWORD and press ENTER'];
+            }
+            $count += 1;
+            return $return_lines;
+    });
+
+    stderr_like {
+        $s3270_console->connect_and_login();
+    } qr/connect_and_login:.*RECONNECT.*trying hard shutdown and reconnect.*/s, 'Reconnect attempted';
```

Comment:
How can I rewrite this test such that it should capture `carp` output written in a for loop?
`connect_and_login();` is producing stderr output from https://github.com/os-autoinst/os-autoinst/blob/master/consoles/s3270.pm#L370 and https://github.com/os-autoinst/os-autoinst/blob/master/consoles/s3270.pm#L385 two times